Mention unit switch statements in conditional parsing docs. (#1487)
zeek/spicy: doc/programming/parsing.rst

    :exec: printf '\01\02\03\04' | spicy-driver %INPUT; printf '\02\02\03\04' | spicy-driver %INPUT
    :show-with: foo.spicy

> For certain cases of conditional parsing unit :ref:`parse_switch` statements
> can be used and should allow more compact and easier to maintain code.

Reviewer: Just word smithing:

```suggestion
For repeated cases of conditional parsing, unit :ref:`parse_switch` statements
might allow for more compact and easier to maintain code.
```
